### PR TITLE
Slime Mutation Changes +new reactions

### PR DESCRIPTION
**Pull request by ghost**
Slime mutation chance starts at a flat 30%

Slime mutation chance is now hereditary, with the new baby slimes
inheriting the mutation chance of the parent (with a variation of -5 to
+5). This allows you to cultivate slimes that are more likely to mutate
over time rather than praying to the RNG forever.

Plasma no longer causes slimes to have a higher mutation chance, and
epinephrine no longer lowers mutation chance.

Red slime extract+plasma now gives a potion that increases mutation
chance. Slime glycerol is no longer obtainable but I don't think I've ever seen anyone use it anyway.

Blue slime extract+blood now gives a potion that lowers mutation chance.

The numbers might need tweaking if this makes xenobio too fast, but
right now it is often frustrating and slow. It is more reliable and
faster to build a mech than breed slimes given the current mutation
system. Hopefully red and blue slimes will be useful now as well.

No Sepia slime fix yet, sorry.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/slime/life.dm [hunk 1]
```diff
@@ -147,7 +147,7 @@
 	temp_change = (temperature - current)
 	return temp_change
 
-/mob/living/simple_animal/slime/handle_chemicals_in_body()
+/*/mob/living/simple_animal/slime/handle_chemicals_in_body()
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/slime/life.dm]: don't comment out code, just removed it, git stores backups should we ever want to revert.

File: code/modules/research/xenobiology/xenobiology.dm [hunk 2]
```diff
@@ -256,6 +263,58 @@
...
 
+/obj/item/weapon/slimestabilizer
+	name = "slime stabilizer"
+	desc = "A potent chemical mix that will reduce the chance of a slime mutating."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "bottle15"
+
+/obj/item/weapon/slimestabilizer/attack(mob/living/simple_animal/slime/M as mob, mob/user as mob)
+	if(!isslime(M))
+		user << "<span class='warning'> The stabilizer only works on baby slimes!</span>"
```
Comment (Anonus) [inline code/modules/research/xenobiology/xenobiology.dm]: It's not just baby slimes though

File: code/modules/research/xenobiology/xenobiology.dm [hunk 3]
```diff
@@ -256,6 +263,58 @@
...
+
+/obj/item/weapon/slimemutator
+	name = "slime mutator"
+	desc = "A potent chemical mix that will increase the chance of a slime mutating."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "bottle3"
+
+/obj/item/weapon/slimemutator/attack(mob/living/simple_animal/slime/M as mob, mob/user as mob)
+	if(!isslime(M))
+		user << "<span class='warning'> The mutator only works on baby slimes!</span>"
```
Comment (Anonus) [inline code/modules/research/xenobiology/xenobiology.dm]: See above

File: code/modules/mob/living/simple_animal/slime/life.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/slime/life.dm]: Old habit from google code, sorry

File: code/modules/research/xenobiology/xenobiology.dm [hunk 4]
```diff
@@ -256,6 +263,58 @@
...
+		return ..()
+	if(M.stat)
+		user << "<span class='warning'> The slime is dead!</span>"
+		return..()
+	if(M.mutator_used == 1)
```
Comment (RemieRichards) [inline code/modules/research/xenobiology/xenobiology.dm]: swap this for ```if(M.mutator_used)``` 

alternatively, make it a proper boolean with
```mutator_used = FALSE```
and then this check remaining ```if(M.mutator_used)

File: code/modules/research/xenobiology/xenobiology.dm [hunk 5]
```diff
@@ -256,6 +263,58 @@
...
+		user << "<span class='warning'> This slime has already consumed a mutator, any more would be far too unstable!</span>"
+		return..()
+	if(M.mutation_chance == 100)
+		user <<"<span class='warning'> The slime is already guaranteed to mutate!</span>"
+		return..()
+
+	user <<"<span class='notice'>You feed the slime the mutator. It is now more likely to mutate.</span>"
+	M.mutation_chance += 5
+	if(M.mutation_chance < 0)
```
Comment (RemieRichards) [inline code/modules/research/xenobiology/xenobiology.dm]: replace this line, the one above it and the one beneath it with
```M.mutation_chance = max(M.mutation_chance+5,0)```

File: code/modules/mob/living/simple_animal/slime/life.dm (hunk 1 above)
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/slime/life.dm]: Yeah it's no biggy, I figured that was your reasoning, no worries.

File: code/modules/mob/living/simple_animal/slime/powers.dm [hunk 6]
```diff
@@ -191,6 +191,7 @@
 				if(i != 1) step_away(M,src)
 				M.Friends = Friends.Copy()
 				babies += M
+				M.mutation_chance = mutation_chance+(rand(-5,5))
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/slime/powers.dm]: ```M.mutation_chance = Clamp(mutation_chance+(rand(5,-5)),0,100)```

File: code/modules/research/xenobiology/xenobiology.dm [hunk 7]
```diff
@@ -256,6 +263,58 @@
...
+		return ..()
+	if(M.stat)
+		user << "<span class='warning'> The slime is dead!</span>"
+		return..()
+	if(M.mutation_chance == 0)
+		user <<"<span class='warning'> The slime already has no chance of mutating!</span>"
+		return..()
+
+	user <<"<span class='notice'>You feed the slime the stabilizer. It is now less likely to mutate.</span>"
+	M.mutation_chance -= 15
```
Comment (RemieRichards) [inline code/modules/research/xenobiology/xenobiology.dm]: ```M.mutation_chance = Clamp(M.mutation_chance,0,100)```

**Comment by Man-Shroom** [comment]
Red/Blue slime extract should have these new effects if combined with water, and keep their old effects.

**Comment by RemieRichards** [comment]
Also ```KorPhaeron:master``, you shouldn't use your master branch for Pull requests, it makes your life harder (doesn't affect us at all)

File: code/modules/research/xenobiology/xenobiology.dm (hunk 2 above)
Comment (KorPhaeron) [inline code/modules/research/xenobiology/xenobiology.dm]: Whoops. Not sure how I managed that. I mean I know what I did but still.

**Comment by AlexCrimson** [comment]
I dont think this change will do anything. People are not gonna get red/blue slimes just so they can get some reagent to lower/increase mutation chance. Its easy enough to get the specific slimes you want without dealing with mutation chance.

Just make water lower the chance, and mutagen increase it.

**Comment by Man-Shroom** [comment]
the above would work too

**Comment by KorPhaeron** [comment]
I thought about mutagen, but it's more or less in infinite supply.

**Comment by AlexCrimson** [comment]
Thats kinda the point. Making it a hard to get chemical will just make it a useless feature nobody ever uses. Right now Xeno does perfectly fine and never has to inject a single slime with plasma or the other stuff to change mutation chance. Making it an ever rarer chemical sure as hell wont suddenly get people to use it.

File: code/modules/research/xenobiology/xenobiology.dm (hunk 7 above)
Comment (KorPhaeron) [inline code/modules/research/xenobiology/xenobiology.dm]: M.mutation_chance = Clamp(M.mutation_chance-15,0,100) work?

Or does clamp only do positive values?

Comment (RemieRichards) [inline code/modules/research/xenobiology/xenobiology.dm]: Oh yeah, I meant to include the -15 in the clamp statement.

```Clamp(value, minimum, maximum)```
it puts the number ```value``` between ```minimum``` and ```maximum```
So if you range includes negatives (Clamp(health, -100, 100)) then it handles them.

**Comment by KorPhaeron** [comment]
I wanted the +mutation to be a bit harder to get now that the bonus will persist between generations.

If I made mutagen give +mutation chance they could dump 50 units into the starting slime and all the children would have a crazy mutation chance as well.

**Comment by AlexCrimson** [comment]
Nobody will bother with Red/Blue slimes just so they can mutate slimes faster. It would take less time to aim for certain slimes compared to getting Red slimes and engineering slimes with high mutation chance.

If its what you want to do then fine, but nobody is going to use it.

**Comment by KorPhaeron** [comment]
I think you're correct after thinking further on it. I usually try and err on making my stuff underpowered but I think I may have made the values too low (both starting flat chance and the mutator bonus) for the new system to actually be helpful. I'll tweak it upwards to make red slimes an attractive alternative to try for if you happen to get orange on the first split.

**Comment by AlexCrimson** [comment]
Eh that would probably be a good solution. Still think the reaction should only require water though. Plasma seems too valuable to waste converting into this stuff.

**Pull request merged**